libsolidity/AST.cpp

```diff
@@ -673,6 +676,106 @@ void Identifier::checkTypeRequirements()
 		BOOST_THROW_EXCEPTION(createTypeError("Declaration referenced before type could be determined."));
 }
 
+void FunctionIdentifier::overloadResolution()
+{
+	solAssert(!m_overloadedDeclarations.empty(), "FunctionIdentifier not resolved.");
+	solAssert(!m_referencedDeclaration, "Referenced declaration should be null before overload resolution.");
+
+	if (m_overloadedDeclarations.size() == 1)
+	{
+		setReferencedDeclaration(**m_overloadedDeclarations.begin());
+		return;
+	}
+
+	bool resolved = false;
+	auto const& arguments = getFunctionCall().getArguments();
+	auto const& argumentNames = getFunctionCall().getNames();
+
+	if (argumentNames.empty())
+	{
+		// positional arguments
+		for (auto&& declaration: m_overloadedDeclarations)
+		{
+			auto type = declaration->getType();
+			FunctionType const& functionType = dynamic_cast<FunctionType const&>(*type);
+
+			auto const& parameterTypes = functionType.getParameterTypes();
+
+			if (arguments.size() == parameterTypes.size())
+			{
+				bool ok = true;
+				for (size_t i = 0; ok && i < arguments.size(); i++)
+					if (!functionType.takesArbitraryParameters() &&
+						!arguments[i]->getType()->isImplicitlyConvertibleTo(*parameterTypes[i]))
+						ok = false;
+				if (ok)
+				{
+					if (resolved)
+						BOOST_THROW_EXCEPTION(createTypeError("Ambiguous function call"));
+					resolved = true;
+					setReferencedDeclaration(*declaration);
+				}
+			}
+		}
+	}
+	else
+	{
+		// named arguments
+		for (auto&& declaration: m_overloadedDeclarations)
+		{
+			auto type = declaration->getType();
+			FunctionType const& functionType = dynamic_cast<FunctionType const&>(*type);
+
+			auto const& parameterTypes = functionType.getParameterTypes();
+			auto const& parameterNames = functionType.getParameterNames();
+
+			// to use named arguments, parameter names can't be omitted
+			if (parameterTypes.size() == parameterNames.size() && parameterTypes.size() == arguments.size())
+			{
+				bool ok = true;
+				for (auto const& parameterName: parameterNames)
+				{
+					bool found = false;
+					for (size_t i = 0; !found && i < argumentNames.size(); i++)
+						if ((found = (parameterName == *argumentNames[i])))
+						{
+							// we found the actual parameter position
+							// positionedArguments.push_back(arguments[i]);								
+						}
+				}
+
+
+			}
+		}
+	}
+
+	if (!resolved)
+		BOOST_THROW_EXCEPTION(createTypeError("Can't resolve identifier"));
+}
+
+void FunctionIdentifier::checkTypeRequirements()
+{
+	std::vector<Declaration const*> declarations{ m_overloadedDeclarations.cbegin(), m_overloadedDeclarations.cend() };
+
+	if (declarations.size() > 1)
+		for (auto&& declaration: declarations)
+			if (dynamic_cast<FunctionDefinition const*>(declaration) == nullptr)
+				BOOST_THROW_EXCEPTION(createTypeError("Only functions allow multiple declarations."));
```

Comment:
That should actually be an error of `declaration` and should also be checked somewhere else (for example, it is still an error, even if the declarations are not used).